ladybug_geometry/polyskel.py

```diff
@@ -0,0 +1,558 @@
+				# potentially parallel edge)
+
+				# Make normalized copies of vectors
+				norm_edge_left_v  = Vector2D.normalize(self.edge_left.v)
+				norm_edge_right_v = Vector2D.normalize(self.edge_right.v)
+				norm_edge_v = Vector2D.normalize(edge.edge.v)
+
+				# Compute dot
+				leftdot  = abs(norm_edge_left_v.dot(norm_edge_v))
+				rightdot = abs(norm_edge_right_v.dot(norm_edge_v))
+				selfedge = self.edge_left if leftdot < rightdot else self.edge_right
+				otheredge = self.edge_left if leftdot > rightdot else self.edge_right
+
+				# Make copies of edges and compute intersection
+				self_edge_copy = LineSegment2D(selfedge.p, selfedge.v)
+				edge_edge_copy = LineSegment2D(edge.edge.p, edge.edge.v)
+				i = intersection2d.intersect_line2d(self_edge_copy, edge_edge_copy)
+				if i is not None and not _approximately_equals(i, self.point):
+					#locate candidate b
+					linvec = Point2D.normalize(self.point - i)
+					edvec  = Point2D.normalize(edge.edge.v)
+					if linvec.dot(edvec)<0:
+						edvec = -edvec
+
+					bisecvec = edvec + linvec
+					if abs(bisecvec) == 0:
+						continue
+					bisector = LineSegment2D(i, bisecvec)
+					b = bisector.intersect(self.bisector)
+
+					if b is None:
+						continue
+
+					#check eligibility of b
+					# a valid b should lie within the area limited by the edge and the bisectors of its two vertices:
+					xleft = _determinant(edge.bisector_left.v.normalized(), (b - edge.bisector_left.p).normalized()) > 0
+					xright = _determinant(edge.bisector_right.v.normalized(), (b - edge.bisector_right.p).normalized()) < 0
+					xedge = _determinant(edge.edge.v.normalized(), (b - edge.edge.p).normalized()) < 0
+
+					if not (xleft and xright and xedge):
+						log.debug("\t\tDiscarded candidate %s (%s-%s-%s)", b, xleft, xright, xedge)
+						continue
+
+					log.debug("\t\tFound valid candidate %s", b)
+					events.append( _SplitEvent(Line2(edge.edge).distance(b), b, self, edge.edge) )
+
+		# Intersect line2d with line2d (does not assume lines are infinite)
+		i_prev = intersection2d.intersect_line2d(self.bisector, self.prev.bisector)
+		i_next = intersection2d.intersect_line2d(self.bisector, self.prev.bisector)
+
+		# Calc distances from edge to intersection
+		dist_to_i_prev = LineSegment2D(
+			self.edge_left.p.duplicate(),
+			self.edge_left.v.duplicate())
+		dist_to_i_prev = dist_to_i_prev.distance_to_point(i_prev)
+		dist_to_i_next = LineSegment2D(
+			self.edge_right.p.duplicate(),
+			self.edge_right.v.duplicate())
+		dist_to_i_next = dist_to_i_next.distance_to_point(i_next)
+
+		# Make EdgeEvent and append to events
+		if i_prev is not None:
+			events.append(_EdgeEvent(dist_to_i_prev, i_prev, self.prev, self))
+		if i_next is not None:
+			events.append(_EdgeEvent(dist_to_i_next, i_next, self, self.next))
+
+		if not events:
+			return None
+
+		ev = min(
+			events,
+			key=lambda event: self.point.distance_to_point(event.intersection_point)
```

Comment:
Why not using list comprehension. It is cleaner and easier to read and understand.